demos/instr.h

```diff
@@ -182,38 +177,76 @@ inline void BoundsCheck(const char *str, size_t offset) {
   asm volatile(".byte 0x62, 0x02"::"a"(offset), "d"(&string_bounds):"memory");
 }
 
-// Reads an offset from the FS segment.
+// Reads an offset from the SS segment.
 // Must be inlined because the fault occurs inside and the stack pointer would
 // be shifted.
 // We fetch offset + 1 from the segment base, because the base is shifted one
 // byte below to bypass 1-byte minimal segment size.
 SAFESIDE_ALWAYS_INLINE
-inline unsigned int ReadUsingFS(unsigned int offset) {
+inline unsigned int ReadUsingSS(unsigned int ss, unsigned int offset) {
   unsigned int result;
 
   asm volatile(
-      "movzbl %%fs:(, %1, 1), %0\n"
-      :"=r"(result):"r"(offset + 1):"memory");
+      "movl %1, %%ss\n"
```

Comment:
A few things on the inline assembly, here and below:
- Do we need the "memory" clobber?
- Instead of specifying specific registers, let's just use `"r"` and give the compiler freedom to choose the register. Some experiments on Compiler Explorer suggest GCC and Clang will respect any constraints of the individual instructions.
- Let's also remove the data size suffixes. `xor` works just as well as `xorl`.